.github/workflows/main.yml

```diff
@@ -89,25 +99,23 @@ jobs:
         run: |
           source workflow_scripts/protobuf/build_protobuf_unix.sh 3 $(pwd)/protobuf/protobuf_install
 
-      - name: Set up MSBuild
-        if: matrix.os == 'windows-latest'
+      - name: Set up MSBuild (arm64)
+        if: startsWith(matrix.os,'windows-11-arm')
+        uses: microsoft/setup-msbuild@6fb02220983dee41ce7ae257b6f4d8f9bf5ed4ce # v2.0.0
+        with:
+          msbuild-architecture: arm64
+
+      - name: Set up MSBuild (x64)
+        if: startsWith(matrix.os,'windows-latest')
         uses: microsoft/setup-msbuild@6fb02220983dee41ce7ae257b6f4d8f9bf5ed4ce # v2.0.0
         with:
           msbuild-architecture: x64
 
       - name: Install external protobuf - Windows
-        if: matrix.protobuf_type == 'External' && matrix.os == 'windows-latest'
+        if: matrix.protobuf_type == 'External' && startsWith(matrix.os, 'windows')
         run: |
           workflow_scripts/protobuf/build_protobuf_win.ps1
```

Comment:
Do we need to provide the arch parameter here?

Reply:
we could. Otherweise arch would be a system variable?